src/D2L.CodeStyle.Analyzers/Common/MutabilityInspector.cs

```diff
@@ -34,10 +35,20 @@ public sealed class MutabilityInspector {
 			"System.Type",
 			"System.Uri",
 			"System.String",
-			"System.StringComparer",
 			"System.Xml.Serialization.XmlSerializer"
 		}.ToImmutableHashSet();
 
+		/// <summary>
+		/// A list of marked immutable types owned externally.
+		/// </summary>
+		private static readonly ImmutableHashSet<string> MarkedImmutableTypes = new HashSet<string> {

```

Comment:
We call these out separately because it allows us to do things like this:

```
class Foo : StringComparer { ... } // Analyzer will ensure that `Foo` is immutable, because it's marked immutable
```
Then we can use `Foo` anywhere and be happy.